OSLO integration work - update all lang term names

imcdonald-d2l opened the pull request:
Updating **all** language terms in CPD views so they follow the desired naming convention:

- `{cpdView}\{termName}` if term name only exists on a specific view
- `{shared}\{termName}` if term is used across multiple views
-  `{shared}\{sharedComponent}\{termName}` if term is used exclusively in a shared component that is used in multiple views

**Note:** we have to do double `\\` in code to escape the backslash character or the code will break.

Doing this to follow to the recommended naming convention outlined in OSLO documentation [here](https://docs.dev.d2l/index.php/Oslo#Follow_the_OSLO_langterm_naming_convention).

This naming convention will help us sort and search OSLO terms when updating them in the language management tool.

IhabAlkabary commented:
Overall note: The old lang term files, including the translations, are ordered in ascending order. Do we need to maintain this sort order with OSLO?

imcdonald-d2l commented:
> Overall note: The old lang term files, including the translations, are ordered in ascending order. Do we need to maintain this sort order with OSLO?

OSLO documentation does not say anything about the order of lang terms so I expect this not to be an issue. Activity-editor uses OSLO and their lang terms are not sorted as well. See `en.js` file [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/lang/en.js).

imcdonald-d2l commented:
A better approach to reviewing this PR is probably to override the FRA on a test site or locally and make sure no lang terms are messed up and the FRA functions as expected. Reviewing in detail is basically impossible given it's size.

lweber-d2l commented:
Just curious why each of the langterm files has 13 fewer lines than before.  Were some unused langterms identified and removed?

imcdonald-d2l commented:
> Just curious why each of the langterm files has 13 fewer lines than before. Were some unused langterms identified and removed?

Yes, that's exactly it. I removed unused lang terms. That work is in commits `lang term cleanup part 1` and `remove unused lang terms part 2`

IhabAlkabary commented on locales/ar-SA.json:
```diff
@@ -1,148 +1,136 @@
+   "shared\\cancelText" : "إلغاء",
+   "shared\\ariaChooseChoice" : "اختيار {choice}",
+   "shared\\subjectProgress\\combinedText" : "مجمّعة",
+   "shared\\subjectProgress\\completedText" : "Completed:",
```
Looks like an existing bug with the term not translated here.

IhabAlkabary replied:
Maybe we should contact the Translation team about this

imcdonald-d2l replied:
Yeah, hopefully it is caught in next round of serge translations. If not we can reach out to translations team

lweber-d2l replied:
Assuming this missing term is not a blocker for merging, I approve this base on inspection of the changes.  I didn't try the FRA override as part of review but will check each page of the tool as part of QA.
